### PR TITLE
chore: home page and setup done

### PR DESCRIPTION
**Pull request by Mohitkumar33**
Added the routes to all the pages.
Made the home. Page
Called the Apis of Home page and categories

### CODE REVIEW

File: src/contexts/allContexts/main-context.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import { setCategories, setVideos } from "../../utilities/categoriesUtils";
+import { reducerFunction } from "./reducerFunction";
+
+const context = createContext(null);
+
+const ContextProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(reducerFunction, {
+    allVideos: [],
+    allCategories: [],
+    // likes: [],
```
Comment (deepak29-git) [inline src/contexts/allContexts/main-context.js:12]: remove commented code

File: src/contexts/allContexts/reducerFunction.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      return {
+        ...state,
+        allCategories: action.payload,
+      };
+    case "SET_VIDEOS":
+      return {
+        ...state,
+        allVideos: action.payload,
+      };
+    // case "SET_LIKES":
```
Comment (deepak29-git) [inline src/contexts/allContexts/reducerFunction.js:13]: remove unnecessary code

**Review by deepak29-git — APPROVED** [review]

File: src/utilities/categoriesUtils.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import axios from "axios";
+
+const setCategories = async () => {
+  try {
+    const { data } = await axios.get("/api/categories");
+    return data.categories;
```
Comment (Richa-rani02) [inline src/utilities/categoriesUtils.js]: suggestion: you can extract all the Api separately

**Review by Richa-rani02 — APPROVED** [review]
Remove all the commented code .everything else looks fine

**Pull request merged**